.github/workflows/build-push-images.yaml

```diff
@@ -220,4 +232,84 @@ jobs:
+          echo "::set-output name=vulnerability_scan_pushed::true"
+
+      # Tag images with the latest commit ID if they are not changed.
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-access-setup
+        if: steps.build-image-access-setup.outputs.access_setup_pushed != 'true'
+        env:
+          image: access-setup
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-cluster-setup
+        if: steps.build-image-cluster-setup.outputs.cluster_setup_pushed != 'true'
+        env:
+          image: cluster-setup
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-gateway-deployment
+        if: steps.build-image-gateway-deployment.outputs.gateway_deployment_pushed != 'true'
+        env:
+          image: gateway-deployment
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-kcp-registrar
+        if: steps.build-image-access-setup.outputs.kcp_registrar_pushed != 'true'
+        env:
+          image: kcp-registrar
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-devenv
+        if: steps.build-image-access-setup.outputs.devenv_pushed != 'true'
+        env:
+          image: devenv
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-shellcheck
+        if: steps.build-image-access-setup.outputs.shellcheck_pushed != 'true'
+        env:
+          image: shellcheck
+          registry: quay.io/redhat-pipeline-service
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_TOKEN }}
+        run: |
+          ./images/quay-upload/image-upload.sh
+
+      - name: Tag latest commit ID to quay.io
+        id: tag-commit-quay-vulnerability-scan
+        if: steps.build-image-vulnerability-scan.outputs.vulnerability_scan_pushed != 'true'
```

Comment:
`vulnerability_scan_pushed` is defined in the "Print image url" step
But here you are referring to output from "Build vulnerability Image" step. 
You have tested it and it works this way ?
(same for all steps)

Reply:
You're right. I've changed the if condition check if check with `steps.filter.outputs.access-setup`.
Only if the paths are changed would it trigger image build and push. So, if this is not true, I am fetching the latest commit on Git, tagging it and pushing it to Quay.